covid/extract.py

```diff
@@ -44,3 +45,59 @@ def get_state_abbreviations_to_names():
         abbreviations = json.load(state_abbreviations_file)
 
     return abbreviations
+
+
+def power_bi_extractor(response):
+    data = json.loads(response.text)
+    value_list = data['results'][0]['result']['data']['dsr']['DS'][0]['PH'][1]['DM1']
+    for vl in value_list:
+        yield vl['C']
+
+
+# CDC data source: https://www.cdc.gov/nhsn/covid19/report-patient-impact.html
+
+def extract_cdc_inpatient_beds():
+    # State Representative Estimates for Percentage of Inpatient Beds Occupied (All Patients)
+    response = requests.post(
+        ugc.URL,
+        headers={**ugc.BASE_HEADERS, **ugc.INPATIENT_BED_HEADERS},
+        data=open("./covid/extract_config/inpatient_bed_query.json"))
+
+    df = pd.DataFrame(
+        power_bi_extractor(response),
+        columns=["State", "inpatient_bed_percent_occupied", "inpatient_beds_occupied"])
+
+    df = df.set_index('State')
+
+    return df
+
+
+def extract_cdc_icu_beds():
+    # State Representative Estimates for Percentage of ICU Beds Occupied (All Patients)
+    response = requests.post(
+        ugc.URL,
+        headers={**ugc.BASE_HEADERS, **ugc.ICU_BED_HEADERS},
+        data=open("./covid/extract_config/icu_bed_query.json"))
+
+    df = pd.DataFrame(
+        power_bi_extractor(response),
+        columns=["State", "icu_percent_occupied", "icu_beds_occupied"])
+
+    df = df.set_index('State')
+
+    return df
```

Comment:
Amazing! @lucasmbrown my suggested method to handle the time-series nature is to add a "date" column like we have for the tracker data and then upload this to a sheet for storage. @BriceWolfgang -- does the BI response include any info on when the data were last updated? Can we scrape that somehow?

What the Category 3 sheet would then look like is:

|State|Date|% ICU Occupied| # ICU Occupied | % Inpatient Occupied| # Inpatient Occupied| |% Facilities Reporting | # Facilities Reporting| 
|Alabama | 2020-07-06| 50% | 500 | 60% | 600 | 15% | 20
| Alabama | 2020-07-09| ... | ... | ...| ... | ...|

Reply:
Time has been stamped